.github/ISSUE_TEMPLATE/collaboration_proposal.yml

```diff
@@ -0,0 +1,43 @@
+name: Collaboration proposal
+description: Propose a collaboration with OpenRewrite
+title: "Proposal to collaborate with <your project>"
+labels: [access]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        At OpenRewrite we invite anyone to [contribute to our projects](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#ways-to-contribute).
+        
+        Collaborations allow us to work closely together with other communities, and provide write access to our repositories to trusted members of those communities.
+        We already collaborate with [many organizations and individuals](https://github.com/orgs/openrewrite/repositories?q=&type=public&language=&sort=stargazers), and we're always looking for new communities to work together.
+        
+        If you would like to propose a collaboration with OpenRewrite, please fill out the following information.
```

Comment:
I think that we should explain here what does it mean that we collaborate. What are the benefits and the rules. If there is too much information here, we should create some docs and link them. WDYT?

Reply:
Agree with you there; didn't want to overwhelm here, or potentially limit what collaborations proposals we get in by specifying too much up front. For instance a collaboration would look vastly different for helping maintain Apache Maven as it would for migrating Helidon users. Best to [document some proposed scenarios](https://github.com/openrewrite/rewrite-docs/issues/188#issuecomment-1613205315) more extensively in the docs, and then link those here.

Reply:
Leaning towards a merge of this first, such that we can link here from the documentation, then revisit to add a link with more details to the form once the documentation itself is up.